docs/providers.rst

```diff
@@ -180,7 +185,7 @@ IPCProvider
 AsyncHTTPProvider
 ~~~~~~~~~~~~~~~~~
 
-.. py:class:: web3.providers.async_rpc.AsyncHTTPProvider(endpoint_uri[, request_kwargs])
+.. py:class:: web3.providers.rpc.AsyncHTTPProvider(endpoint_uri, request_kwargs={}, exception_retry_configuration=ExceptionRetryConfiguration())
```

Comment:
If it's none we set it to `{}` in the `__init__()` though... do we think it's better to hint at the type here by using `{}` or to be technically correct and use `None` even though that it doesn't give a good type hint?

Reply:
We can use `None` and clarify below that it sets it to an empty dict? I'm ok with either approach.

Reply:
Eh, I guess since it's just in the docs, we can leave it as a dict. I'll leave it up to you :)

Reply:
I'll leave my reasoning for keeping it here. We have to do a dance for `ExceptionRetryConfiguration` as well where we use `Empty` since we want the ability to set it to `None` to turn it off. So I think if we go the technically correct route we'd have to put `Empty` as the default value for `ExceptionRetryConfiguration` and then explain why and I think that complicates everything more than any user cares about 😆 .

I'm going to opt to use values that if you don't touch the configuration at all, those are what we will hard code in the `__init__()`, which feels more valuable to a user than what the signature represents.